src/main/java/dev/j3fftw/litexpansion/machine/Recycler.java

```diff
@@ -54,13 +56,25 @@ public Recycler() {
 
     private void setupInv() {
         createPreset(this, "&8Recycler", blockMenuPreset -> {
-            for (int i = 0; i < 27; i++)
+            for (int i : BORDER)
                 blockMenuPreset.addItem(i, ChestMenuUtils.getBackground(), ChestMenuUtils.getEmptyClickHandler());
 
-            blockMenuPreset.addItem(INPUT_SLOT, null, (player, i, itemStack, clickAction) -> true);
             Utils.putOutputSlot(blockMenuPreset, OUTPUT_SLOT);
 
             blockMenuPreset.addItem(PROGRESS_SLOT, new CustomItem(Material.DEAD_BUSH, "&7Progress"));
+            blockMenuPreset.addMenuClickHandler(PROGRESS_SLOT, ChestMenuUtils.getEmptyClickHandler());
+        });
+
+        registerBlockHandler(getId(), (p, b, stack, reason) -> {
+            BlockMenu inv = BlockStorage.getInventory(b);
+            Location location = b.getLocation();
+
+            if (inv != null) {
+                inv.dropItems(location, getInputSlots());
+                inv.dropItems(location, getOutputSlots());
+            }
+
+            return true;
```

Comment:
make it a method because soon LX will have more machines that needs to use this anyways